### PR TITLE
Retain brush selection on plot resize

### LINKED ISSUE #174: [Bug]: Selected brush gets cleared on resizing because of re-render of plot
### What happened?

Since we re-render the plots when resizing the selected brush gets reset when we do this. It does not seem like a trivial task but it will certainly improve the UX.

Please refer to the GIF for an example:
Kapture 2023-08-04 at 21 09 04

### sessionInfo()

```R
R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] C/UTF-8/C/C/C/C

time zone: Asia/Kolkata
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] teal.slice_0.3.0.9028             testthat_3.1.9                   
 [3] sparkline_2.0                     teal.modules.general_0.2.15.9033 
 [5] shinyTree_0.2.7                   ggmosaic_0.3.3                   
 [7] ggplot2_3.4.2                     teal.modules.clinical_0.8.15.9013
 [9] tern_0.8.5.9004                   rtables_0.6.2.9001               
[11] formatters_0.5.1.9001             teal.goshawk_0.1.14.9020         
[13] teal_0.13.0.9025                  teal.transform_0.3.0.9009        
[15] teal.data_0.2.0.9007              goshawk_0.1.14.9016              
[17] stringr_1.5.0                     scda.2022_0.1.5.9000             
[19] scda_0.1.6.9010                   dplyr_1.1.2                      
[21] magrittr_2.0.3                    DescTools_0.99.49                
[23] shiny_1.7.4.1                    

loaded via a namespace (and not attached):
  [1] rstudioapi_0.14          jsonlite_1.8.7           estimability_1.4.1      
  [4] farver_2.1.1             rmarkdown_2.23           geepack_1.3.9           
  [7] fs_1.6.3                 vctrs_0.6.3              memoise_2.0.1           
 [10] usethis_2.2.1            htmltools_0.5.5          forcats_1.0.0           
 [13] broom_1.0.5              cellranger_1.1.0         sass_0.4.7              
 [16] bslib_0.5.0              desc_1.4.2               htmlwidgets_1.6.2       
 [19] fontawesome_0.5.1        emmeans_1.8.7            rootSolve_1.8.2.3       
 [22] plotly_4.10.2            cachem_1.0.8             mime_0.12               
 [25] lifecycle_1.0.3          teal.widgets_0.3.0.9007  pkgconfig_2.0.3         
 [28] Matrix_1.5-4.1           R6_2.5.1                 fastmap_1.1.1           
 [31] rbibutils_2.2.13         digest_0.6.33            Exact_3.2               
 [34] colorspace_2.1-0         ggnewscale_0.4.9         ps_1.7.5                
 [37] shinycssloaders_1.0.0    rprojroot_2.0.3          pkgload_1.3.2           
 [40] crosstalk_1.2.0          teal.code_0.3.0.9009     labeling_0.4.2          
 [43] fansi_1.0.4              httr_1.4.6               compiler_4.3.1          
 [46] remotes_2.4.2            proxy_0.4-27             withr_2.5.0             
 [49] backports_1.4.1          logger_0.2.2             pkgbuild_1.4.2          
 [52] R.utils_2.12.2           MASS_7.3-60              sessioninfo_1.2.2       
 [55] gld_2.6.6                tools_4.3.1              robslopes_1.1.3         
 [58] httpuv_1.6.11            shinyvalidate_0.1.2      R.oo_1.25.0             
 [61] glue_1.6.2               callr_3.7.3              nlme_3.1-162            
 [64] R.cache_0.16.0           promises_1.2.0.1         grid_4.3.1              
 [67] checkmate_2.2.0          generics_0.1.3           gtable_0.3.3            
 [70] R.methodsS3_1.8.2        class_7.3-22             tidyr_1.3.0             
 [73] data.table_1.14.8        lmom_2.9                 utf8_1.2.3              
 [76] ggrepel_0.9.3            pillar_1.9.0             later_1.3.1             
 [79] splines_4.3.1            tern.gee_0.1.0.9006      lattice_0.21-8          
 [82] survival_3.5-5           tidyselect_1.2.0         miniUI_0.1.1.1          
 [85] knitr_1.43               teal.logger_0.1.1.9009   xfun_0.39               
 [88] expm_0.999-7             brio_1.1.3               devtools_2.4.5          
 [91] DT_0.28                  stringi_1.7.12           lazyeval_0.2.2          
 [94] yaml_2.3.7               boot_1.3-28.1            shinyWidgets_0.7.6      
 [97] evaluate_0.21            tibble_3.2.1             BiocManager_1.30.21     
[100] cli_3.6.1                xtable_1.8-4             Rdpack_2.4              
[103] processx_3.8.2           munsell_0.5.0            jquerylib_0.1.4         
[106] styler_1.10.1            Rcpp_1.0.11              teal.reporter_0.1.1.9020
[109] readxl_1.4.2             parallel_4.3.1           ellipsis_0.3.2          
[112] assertthat_0.2.1         prettyunits_1.1.1        mcr_1.3.2               
[115] profvis_0.3.8            urlchecker_1.0.1         viridisLite_0.4.2       
[118] mvtnorm_1.2-2            scales_1.2.1             e1071_1.7-13            
[121] crayon_1.5.2             purrr_1.0.1              rlang_1.1.1             
[124] shinyjs_2.1.0
```

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by vedhav**
Closes #174

### CODE REVIEW

**Comment by chlebowa** [comment]
Would you mind modifying the last example app while you're at it?
```
shinyApp(
  ui = fluidPage(
    useShinyjs(),
    actionButton("button", "Show/Hide"),
    plot_with_settings_ui(
      id = "plot_with_settings"
    )
  ),
  server = function(input, output, session) {
    plot_r <- reactive(ggplot2::qplot(data = faithful, x = waiting, y = eruptions))

    show_hide_signal_rv <- reactiveVal(TRUE)

    observeEvent(input$button, show_hide_signal_rv(!show_hide_signal_rv()))

    plot_with_settings_srv(
      id = "plot_with_settings",
      plot_r = plot_r,
      height = c(400, 100, 1200),
      width = c(500, 250, 750),
      show_hide_signal = reactive(show_hide_signal_rv())
    )
  }
)
```

**Review by chlebowa — APPROVED** [review]
Great :+1:

File: .lintr [hunk 1]
```diff
@@ -1,5 +1,6 @@
 linters: linters_with_defaults(
   line_length_linter = line_length_linter(120),
   cyclocomp_linter = NULL,
-  object_usage_linter = NULL
+  object_usage_linter = NULL,
+  indentation_linter = NULL
```
Comment (averissimo) [inline .lintr:5]: Why did you add this?

Comment (vedhav) [reply .lintr:5]: The new indentation checks were ridiculous and I could not find a suitable option for [indentation_linter](https://lintr.r-lib.org/reference/indentation_linter.html):
```r
R/panel_group.R:29:10: style: [indentation_linter] Indentation should be 8 spaces but is 10 spaces.
          allTags()
        ~^
R/plot_with_settings.R:606:4: style: [indentation_linter] Indentation should be 14 spaces but is 4 spaces.
    is.null(brush$panelvar2) && is.character(original_panelvar2)) {
   ^~~~~~~~~~~
R/shiny_brush examples.R:4:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    ui = fluidPage(
  ~^
R/shiny_brush examples.R:5:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        sidebarLayout(
    ~~~^
R/shiny_brush examples.R:6:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            sidebarPanel(
      ~~~~~^
R/shiny_brush examples.R:7:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                actionButton("newplot", "New plot")
        ~~~~~~~^
R/shiny_brush examples.R:8:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            ),
      ~~~~~^
R/shiny_brush examples.R:10:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                plotOutput("plot")
        ~~~~~~~^
R/shiny_brush examples.R:11:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            )
      ~~~~~^
R/shiny_brush examples.R:12:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        )
    ~~~^
R/shiny_brush examples.R:13:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    ),
  ~^
R/shiny_brush examples.R:15:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        output$plot <- renderPlot({
    ~~~^
R/shiny_brush examples.R:16:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            input$newplot
      ~~~~~^
R/shiny_brush examples.R:20:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:21:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    }
  ~^
R/shiny_brush examples.R:27:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    ui = basicPage(
  ~^
R/shiny_brush examples.R:28:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        fluidRow(
    ~~~^
R/shiny_brush examples.R:29:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            column(
      ~~~~~^
R/shiny_brush examples.R:30:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                width = 4,
        ~~~~~~~^
R/shiny_brush examples.R:32:20: style: [indentation_linter] Indentation should be 10 spaces but is 20 spaces.
                    height = 300,
          ~~~~~~~~~^
R/shiny_brush examples.R:36:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                ),
        ~~~~~~~^
R/shiny_brush examples.R:41:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            ),
      ~~~~~^
R/shiny_brush examples.R:43:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                width = 4,
        ~~~~~~~^
R/shiny_brush examples.R:46:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            ),
      ~~~~~^
R/shiny_brush examples.R:48:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                width = 4,
        ~~~~~~~^
R/shiny_brush examples.R:51:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            )
      ~~~~~^
R/shiny_brush examples.R:52:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        )
    ~~~^
R/shiny_brush examples.R:53:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    ),
  ~^
R/shiny_brush examples.R:55:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        data <- reactive({
    ~~~^
R/shiny_brush examples.R:56:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            input$newplot
      ~~~~~^
R/shiny_brush examples.R:59:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:61:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            d <- data()
      ~~~~~^
R/shiny_brush examples.R:63:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:65:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            cat("Click:\n")
      ~~~~~^
R/shiny_brush examples.R:67:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:69:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            cat("Hover (throttled):\n")
      ~~~~~^
R/shiny_brush examples.R:71:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:73:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            cat("Brush (debounced):\n")
      ~~~~~^
R/shiny_brush examples.R:75:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:77:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            # For base graphics, we need to specify columns, though for ggplot2,
      ~~~~~^
R/shiny_brush examples.R:81:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                return()
        ~~~~~~~^
R/shiny_brush examples.R:82:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            }
      ~~~~~^
R/shiny_brush examples.R:84:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:86:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            res <- brushedPoints(data(), input$plot_brush, "speed", "dist")
      ~~~~~^
R/shiny_brush examples.R:88:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                return()
        ~~~~~~~^
R/shiny_brush examples.R:89:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            }
      ~~~~~^
R/shiny_brush examples.R:91:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:92:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    }
  ~^
R/shiny_brush examples.R:99:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    ui = basicPage(
  ~^
R/shiny_brush examples.R:100:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        fluidRow(
    ~~~^
R/shiny_brush examples.R:101:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            column(
      ~~~~~^
R/shiny_brush examples.R:102:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                width = 4,
        ~~~~~~~^
R/shiny_brush examples.R:104:20: style: [indentation_linter] Indentation should be 10 spaces but is 20 spaces.
                    height = 300,
          ~~~~~~~~~^
R/shiny_brush examples.R:107:24: style: [indentation_linter] Indentation should be 12 spaces but is 24 spaces.
                        id = "image_hover",
            ~~~~~~~~~~~^
R/shiny_brush examples.R:110:20: style: [indentation_linter] Indentation should be 10 spaces but is 20 spaces.
                    ),
          ~~~~~~~~~^
R/shiny_brush examples.R:112:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                )
        ~~~~~~~^
R/shiny_brush examples.R:113:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            ),
      ~~~~~^
R/shiny_brush examples.R:115:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                width = 4,
        ~~~~~~~^
R/shiny_brush examples.R:118:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            ),
      ~~~~~^
R/shiny_brush examples.R:120:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                width = 4,
        ~~~~~~~^
R/shiny_brush examples.R:123:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            )
      ~~~~~^
R/shiny_brush examples.R:124:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        )
    ~~~^
R/shiny_brush examples.R:125:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    ),
  ~^
R/shiny_brush examples.R:127:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        output$image <- renderImage({
    ~~~^
R/shiny_brush examples.R:128:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            input$newimage
      ~~~~~^
R/shiny_brush examples.R:130:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            # Get width and height of image output
      ~~~~~^
R/shiny_brush examples.R:134:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            # Write to a temporary PNG file
      ~~~~~^
R/shiny_brush examples.R:137:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            png(outfile, width = width, height = height)
      ~~~~~^
R/shiny_brush examples.R:141:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            # Return a list containing information about the image
      ~~~~~^
R/shiny_brush examples.R:143:16: style: [indentation_linter] Indentation should be 8 spaces but is 16 spaces.
                src = outfile,
        ~~~~~~~^
R/shiny_brush examples.R:148:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            )
      ~~~~~^
R/shiny_brush examples.R:149:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:151:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            cat("Click:\n")
      ~~~~~^
R/shiny_brush examples.R:153:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:155:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            cat("Hover (throttled):\n")
      ~~~~~^
R/shiny_brush examples.R:157:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:159:12: style: [indentation_linter] Indentation should be 6 spaces but is 12 spaces.
            cat("Brush (debounced):\n")
      ~~~~~^
R/shiny_brush examples.R:161:8: style: [indentation_linter] Indentation should be 4 spaces but is 8 spaces.
        })
    ~~~^
R/shiny_brush examples.R:162:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    }
  ~^
tests/testthat/test-plot_with_settings.R:65:6: style: [indentation_linter] Indentation should be 12 spaces but is 6 spaces.
      ggplot2::geom_point())
     ^~~~~~~
```

**Comment by chlebowa** [comment]
We have PR-blocking indentation checks now? :roll_eyes:
